crates/cli/src/config.rs

```diff
@@ -511,15 +570,69 @@ impl Config {
         Self {
             home: RawConfig::new_with_localhost(),
             home_path,
+            doc: Default::default(),
         }
     }
 
+    /// Returns a preserving copy of [Config].
+    fn doc(&self) -> DocumentMut {
+        let mut doc = self.doc.clone();
+
```

Comment:
nit: can we do something like `let RawConfig { ... } = home` so people don't forget to update this function if they add more fields?

Reply:
Done